lib/stdlib/src/proplists.erl

```diff
@@ -668,3 +669,78 @@ split([], Store, Rest) ->
 
 maps_prepend(Key, Val, Dict) ->
     Dict#{Key := [Val | map_get(Key, Dict)]}.
+
+%% ---------------------------------------------------------------------
+
+%% @doc Converts the property list <code>List</code> to a map.
+%%
+%% Shorthand atom values in <code>List</code> will be expanded to an
+%% association of the form <code>Atom => true</code>. Tuples of the
+%% form <code>{Key, Value}</code> in <code>List</code> will be
+%% converted to an association of the form <code>Key => Value</code>.
+%% Anything else will be silently ignored.
+%%
+%% If the same key appears in <code>List</code> multiple times, the
+%% value of the one appearing nearest to the head of <code>List</code>
+%% will be in the result map, that is the value that would be returned
+%% by a call to <code>get_value/2</code> with this key.
+%%
+%% <p>Example:<pre>
+%% to_map([a, {b, 1}, {c, 2}, {c, 3}])</pre>
+%% returns<pre>
+%% #{a => true, b => 1, c => 2}</pre>
+%% </p>
+
+-spec to_map(List) -> Map when
+      List :: [Shorthand | {Key, Value} | term()],
+      Map :: #{Shorthand => 'true', Key => Value},
+      Shorthand :: atom(),
+      Key :: term(),
+      Value :: term().
+
+to_map(List) ->
+    lists:foldr(
+        fun
+            ({K, V}, M) ->
+                M#{K => V};
+            (T, M) when 1 =< tuple_size(T) ->
+                maps:remove(element(1, T), M);
+            (K, M) when is_atom(K) ->
+                M#{K => true};
+            (_, M) ->
+                M
+        end,
+        #{},
+        List
+    ).
+
+%% @doc Converts the property list <code>List</code> to a map
+%% after applying the normalizations given in <code>Stages</code>.
+%%
+%% @see normalize/2
+%% @see to_map/1
+
+-spec to_map(List, Stages) -> Map when
+      List :: [term()],
+      Stages :: [Operation],
+      Operation :: {'aliases', Aliases}
+                 | {'negations', Negations}
+                 | {'expand', Expansions},
+      Aliases :: [{Key, Key}],
+      Negations :: [{Key, Key}],
+      Expansions :: [{Property :: property(), Expansion :: [term()]}],
+      Map :: #{term() => term()}.
+
+to_map(List, Stages) ->
+    to_map(normalize(List, Stages)).
```

Comment:
Since the last clause in `normalize/2` is `compact(List)`, and that is an annoyingly unnecessary pass before `to_map/1`, I thought about implementing `to_map/2` as a copy of `normalize/2` but with the final clause being `to_map(List)` instead. That would OTOH be an annoying code duplication to save one list rebuild.

I am torn here.  Do as you please with that idea.

Reply:
Hm, I'm undecided too... How about a helper function that does "normalize-without-compact"? `normalize/2` can call it and pass the result into `compact/1`, and `to_map/2` can call it and pass the result into `to_map/1`. What do you think?

Reply:
OTOH, `normalize` itself is pretty lean, probably not worth an extra helper function... Ah well, I'll sleep on it ;)

Reply:
Maybe worth a helper function since that makes it obvious what is common, and impossible to by mistake do some change to just one of the normalizing functions.  Finding a pretty name is harder...

Reply:
It won't be exported, so the name doesn't need to be pretty enough to name children after it ;) Would `apply_stages` do?

Reply:
`apply_stages` - certainly!

Reply:
Done :)